packages/wonder-blocks-labeled-field/src/components/labeled-field.tsx

```diff
@@ -103,23 +120,27 @@ export default function LabeledField(props: Props) {
         testId,
         light,
         description,
-        error,
+        errorMessage,
         labels = defaultLabeledFieldLabels,
     } = props;
 
-    const ids = useUniqueIdWithMock("labeled-field");
-    const uniqueId = id ?? ids.get("id");
+    const generatedUniqueId = React.useId();
+    const uniqueId = id ?? `${generatedUniqueId}-labeled-field`;
     const labelId = `${uniqueId}-label`;
     const descriptionId = `${uniqueId}-description`;
     const fieldId = `${uniqueId}-field`;
     const errorId = `${uniqueId}-error`;
 
+    const isRequired = !!required || !!field.props.required;
+    const hasError = !!errorMessage || !!field.props.error;
+    const isLight = light || field.props.light;
```

Comment:
We also check the field's props to see if LabeledField is required, is in an error state, or is in light mode. 

This will make migration easier too so that we don't need to move `light` or `required` props from the field to the LabeledField component. The `errorMessage` prop would need to be set for LabeledField still since fields don't have this prop currently

Reply:
**question:** Would this work in the case where we use custom fields? I'm not sure if this will fail of a custom field doesn't contain one of these props.

Reply:
Good question! 

If LabeledField and a custom field doesn't have the required/error/light props set, then these values will be `false` or `undefined`! (no issues here)

If a custom field doesn't support these props, then they won't be applied to the component. From testing out LabeledField with a plain `<input>` element, I get warnings around props not being supported or are the wrong type 

![image](https://github.com/user-attachments/assets/4df02c33-1f98-4da9-b9cc-f48cf06940c6)

Note: There is documentation around how LabeledField works best with field components that accept `error` and `required` props (I'll update this to include the light prop also) (`?path=/docs/packages-labeledfield--docs#fields`): 
![image](https://github.com/user-attachments/assets/de09c4c6-dd19-4fa7-9deb-a2e15606dbcf)

One idea is we could see if we can refine the `field` type so that the component must have these props supported.

What do you think?

Reply:
I added a story that shows [LabeledField with a plain input element](https://5e1bf4b385e3fb0020b7073c-ybogfhqdoy.chromatic.com/?path=/story/packages-labeledfield--with-non-wb), let me know what you think!

Reply:
> One idea is we could see if we can refine the field type so that the component must have these props supported.

I think it is fine not to enforce those props in `field` instances (at least not for now). I'm sure there will be cases where we still have reusable UI components in webapp that could be combined with `LabeledField` (e.g. `DatePicker`, `TimePicker`, some deprecated text field/area components). If we get to a point where we can get rid of those deprecated versions, then it would be nice to enforce those props.

Maybe these errors/warnings could go away using the nullish coalescing operator (??)

e.g.
```
const isRequired = !!required || !!field.props.required ?? false;
```

Reply:
> Maybe these errors/warnings could go away using the nullish coalescing operator (??)

This doesn't address it unfortunately! The `required` prop isn't as much of an issue since it is a supported html attribute. It's more of the other props since those are specific to WB components! I was trying to see if there's a way for us to check at runtime if a component supports certain props before we set it. I couldn't find anything around this (let me know though if we have a way to do this!)

Some other options:
1. LabeledField only sets html attributes. LabeledField works best though when the field components can handle their own required/error/light states and their testId. 
2. Use a renderField prop instead so that the props are passed as an argument and consumers could decide how to apply the props to their custom element 
3. We support both the current `field` prop and a `renderCustomField` prop (see #3)
4. We leave things as is with the console warnings and custom components need to support the prop 

Let me know what you think or if you have other ideas!

Reply:
Ahhh got it, thanks for clarifying. I'd say 4 would be best in this case as we still offer support to WB fields (making a good case for trying to use these components).